packages/grafana-flamegraph/src/FlameGraph/FlameGraph.tsx

```diff
@@ -67,118 +64,66 @@ const FlameGraph = ({
   }, [data, sandwichItem]);
 
-  const [sizeRef, { width: wrapperWidth }] = useMeasure<HTMLDivElement>();
-  const graphRef = useRef<HTMLCanvasElement>(null);
-  const [tooltipItem, setTooltipItem] = useState<LevelItem>();
-
-  const [clickedItemData, setClickedItemData] = useState<ClickedItemData>();
-
-  useFlameRender({
-    canvasRef: graphRef,
-    colorScheme,
+  const commonCanvasProps = {
     data,
-    focusedItemData,
-    levels,
-    rangeMax,
     rangeMin,
+    rangeMax,
     search,
+    setRangeMin,
+    setRangeMax,
+    onItemFocused,
+    focusedItemData,
     textAlign,
+    onSandwich,
+    colorScheme,
+    totalProfileTicks,
+    totalProfileTicksRight,
     totalViewTicks,
-    // We need this so that if we have a diff profile and are in sandwich view we still show the same diff colors.
-    totalColorTicks: data.isDiffFlamegraph() ? totalProfileTicks : totalViewTicks,
-    totalTicksRight: totalProfileTicksRight,
-    wrapperWidth,
-  });
-
-  const onGraphClick = useCallback(
-    (e: ReactMouseEvent<HTMLCanvasElement>) => {
-      setTooltipItem(undefined);
-      const pixelsPerTick = graphRef.current!.clientWidth / totalViewTicks / (rangeMax - rangeMin);
-      const { levelIndex, barIndex } = convertPixelCoordinatesToBarCoordinates(
-        { x: e.nativeEvent.offsetX, y: e.nativeEvent.offsetY },
-        levels,
-        pixelsPerTick,
-        totalViewTicks,
-        rangeMin
-      );
-
-      // if clicking on a block in the canvas
-      if (barIndex !== -1 && !isNaN(levelIndex) && !isNaN(barIndex)) {
-        const item = levels[levelIndex][barIndex];
-        setClickedItemData({
-          posY: e.clientY,
-          posX: e.clientX,
-          item,
-          level: levelIndex,
-          label: data.getLabel(item.itemIndexes[0]),
-        });
-      } else {
-        // if clicking on the canvas but there is no block beneath the cursor
-        setClickedItemData(undefined);
-      }
-    },
-    [data, rangeMin, rangeMax, totalViewTicks, levels]
-  );
-
-  const [mousePosition, setMousePosition] = useState<{ x: number; y: number }>();
-  const onGraphMouseMove = useCallback(
-    (e: ReactMouseEvent<HTMLCanvasElement>) => {
-      if (clickedItemData === undefined) {
-        setTooltipItem(undefined);
-        setMousePosition(undefined);
-        const pixelsPerTick = graphRef.current!.clientWidth / totalViewTicks / (rangeMax - rangeMin);
-        const { levelIndex, barIndex } = convertPixelCoordinatesToBarCoordinates(
-          { x: e.nativeEvent.offsetX, y: e.nativeEvent.offsetY },
-          levels,
-          pixelsPerTick,
-          totalViewTicks,
-          rangeMin
-        );
+  };
+  const canvas = levelsCallers ? (
+    <>
+      <div className={styles.sandwichCanvasWrapper}>
+        <div className={styles.sandwichMarker}>
+          Callers
+          <Icon className={styles.sandwichMarkerIcon} name={'arrow-down'} />
+        </div>
+        <FlameGraphCanvas
```

Comment:
Sandwich view is now done by rendering 2 separate graphs so lots of code was moved to that smaller component.